lib/matplotlib/lines.py

```diff
@@ -507,7 +504,11 @@ def set_pickradius(self, d):
         d : float
             Pick radius, in points.
         """
-        self.pickradius = d
+        if not isinstance(d, Number) or d < 0:
```

Comment:
Does a pick radius of 0 make sense? It would trigger only if you hit that pixel. Likely `set_pickradius(0)` might be misunderstood as deactivating picking. Anyway, let's not be overly restrictive. -> ok.

Reply:
I don't think zero makes more or less sense than 0.001, for example (especially given that GUI toolkits will typically only report events with a precision of 1px anyways), so I wouldn't overly worry about that.